modules/slugs/includes/qtranslate-slug-import.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Legacy meta and options from QTS plugin.
+ */
+const QTX_SLUGS_LEGACY_QTS_META_PREFIX    = '_qts_slug_';
+const QTX_SLUGS_LEGACY_QTS_OPTIONS_PREFIX = '_qts_';
+const QTX_SLUGS_LEGACY_QTS_OPTIONS_NAME   = 'qts_options';
+
+/**
+ * Check if slugs meta should be imported from the legacy QTS postmeta and termmeta.
+ *
+ * @return string messages giving details, empty if new meta found or no legacy meta found.
+ */
+function qtranxf_slugs_check_import_qts() {
+    global $wpdb;
+
+    /**
+     * Generic function that counts the slugs meta, legacy (QTS) or new (QTX).
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string $prefix prefix for the meta key
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $count_slugs = function ( $table, $prefix, &$msg ) use ( $wpdb ) {
+        $results = $wpdb->get_var( "SELECT count(*) FROM  $table WHERE meta_key like '$prefix%'" );
+        if ( $results ) {
+            $msg[] = sprintf( __( "Found %s slugs from $table.", 'qtranslate' ), $results );
+        }
+    };
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_META_PREFIX, $msg );
+    if ( ! empty( $msg ) ) {
+        // Found some post/term meta with the new keys, no import to suggest (it can still be done manually).
+        return '';
+    }
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+
+    return empty ( $msg ) ? $msg : implode( '<br>', $msg );
+}
+
+/**
+ * Import slugs meta by duplicating the legacy QTS postmeta and termmeta for QTX.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_import_qts_meta( $db_commit ) {
+    global $wpdb;
+
+    $new_prefix = QTX_SLUGS_META_PREFIX;
+    $old_prefix = QTX_SLUGS_LEGACY_QTS_META_PREFIX;
+
+    /**
+     * Generic function that imports old meta into new ones. All existing new meta are erased first.
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string $colid column name giving the meta id (post_id, term_id)
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $import_meta = function ( $table, $colid, &$msg ) use ( $wpdb, $old_prefix, $new_prefix ) {
+        $results = $wpdb->query( "DELETE FROM $table WHERE meta_key like '$new_prefix%'" );
```

Comment:
Not clear why we want to delete existing new named metadata.
Ref similar comment done for existing new named options.

Reply:
> However it could be possible to do a merge if we do it in two steps: (1) overwrite (update) the meta_value for existing meta_key (new prefix), and (2) insert missing keys (new prefix).

Yes, sounds good IMO

Reply:
The new import now selects only the QTS slugs that have not been imported yet. Existing slugs (after rename) are ignored.

Added also a new option to delete before import. This gives max flexibility in case the admin wants to erase all and restart the import from scratch.

Reply:
Finally opted for (2), only new keys are imported. 
(1) can become problematic because it's hard to understand the purpose of the import. Instead, added a new option to delete before import. If checked, the import will clone all the QTS data. It is only meaningful in case the import is done multiple times.

Reply:
Done. Finally opted for (2), the new import now selects only the QTS slugs that have not been imported yet. Existing slugs (after rename) are ignored. It was a bit tricky as the DB primary key is not the meta key.

Overwrite (1) can become problematic because it's hard to understand the purpose of the import in that case. So there's never any overwrite (SQL REPLACE never applies because the IDs are always different). 

Instead, added a new checkbox to _delete before import_. This gives max flexibility in case the admin wants to erase all and restart the import from scratch. It is only meaningful in case the import is done multiple times. It is to ask the import to clone all the QTS data.